sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcSQLQueryBuilder.scala

```diff
@@ -208,7 +208,7 @@ class JdbcSQLQueryBuilder(dialect: JdbcDialect, options: JDBCOptions) {
 
   // If join has been pushed down, reuse join query as a subquery. Otherwise, fallback to
   // what is provided in options.
-  private def tableOrQuery = joinQuery.getOrElse(options.tableOrQuery)
+  def tableOrQuery: String = joinQuery.getOrElse(options.tableOrQuery)
```

Comment:
If you need to access it from derived suites, but does not want to make it overridable, you can use `protected final`
```suggestion
  def tableOrQuery: String = joinQuery.getOrElse(options.tableOrQuery)
```